ext/cl_loader_layers.asciidoc

```diff
@@ -125,6 +137,12 @@ in the <<layer-queries-table, Layer Queries>> table.
 |====
 | Layer Info | Return Type | Description
 | *CL_LAYER_API_VERSION* | cl_layer_api_version | OpenCL Layer API version
+| *CL_LAYER_NAME* | char[] | OpenCL Layer Name
+| *CL_LAYER_OPTIONS* | cl_layer_option[]
+  | OpenCL Layer Options.
+    The list is terminated by a `{NULL, NULL, NULL, NULL}` entry.
+    Layers can, but are not required to, use the requested storage for
+    backing memory.
```

Comment:
I think we need to document a little more precisely what memory the caller is intended to allocate (and free!) and which memory is owned by the implementation and hence should not be freed.  It might be useful to have some psueudocode showing how to query and print these layer options.

Additionally, I think a caller should be able to compute the number of supported options via `param_value_size_ret`.  Is this the case?  If so, do we need to have a special `{NULL, NULL, NULL, NULL}` entry to denote the list terminator?

Reply:
The idea here is that the layer is able to request either a partial allocation (just the NULL terminated array), a full allocation (the array plus backing memory for the strings), or anything in-between. The user is only required to free the contiguous buffer they allocated. Given that the layer can request space for more than the array, the size requested by the layer cannot be used to determine the array length. We could also add a query to get the option count.

Reply:
This makes sense to me - thanks!